src/debug/debugSession.ts

```diff
@@ -270,11 +270,33 @@ export class DebugSession implements IDebugSession {
         await this.startDebugging(cwd, sessionName, proxyResult.proxyDebugPort, proxyResult.proxyAppPort, async () => {
             proxyResult.proxyProcess.kill();
             if (appName) {
-                await this.cleanupResource(`deployment/${appName}`);
+                kubeChannel.showOutput("The debug session exited.");
+                await this.promptForCleanup(`deployment/${appName}`);
             }
         });
     }
 
+    private async promptForCleanup(resourceId: string): Promise<void> {
+        const autoCleanupFlag = vscode.workspace.getConfiguration("vs-kubernetes")["vs-kubernetes.autoCleanupOnDebugTerminate"];
+        if (autoCleanupFlag) {
+            return await this.cleanupResource(resourceId);
+        }
+        const answer = await vscode.window.showWarningMessage(`Resource ${resourceId} will continue running on the cluster.`, "Clean Up", "Always Clean Up");
+        if (answer === "Clean Up") {
+            return await this.cleanupResource(resourceId);
+        } else if (answer === "Always Clean Up") {
+            // The object returned by VS Code API is readonly, clone it first.
+            const oldConfig = vscode.workspace.getConfiguration("vs-kubernetes");
+            const newConfig = {};
+            for (const key of Object.keys(oldConfig)) {

```

Comment:
Can this not be done using `Object.assign`?  (Asking because I need something similar, and was using `assign` - if this way is better then I should copy it!)

Reply:
Nice, use `Object.assign` is better.